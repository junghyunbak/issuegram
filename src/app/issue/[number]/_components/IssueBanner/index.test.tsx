import { render, screen } from "@testing-library/react";
import { IssueBanner } from ".";
import { issue } from "@/mockData";

const thumbnailTomlString = [
  "+++",
  'thumbnail = "https://fake.url"',
  "+++",
  "",
].join("\n");

it("썸네일 주소가 있을경우, img 태그를 렌더링한다.", () => {
  const { container } = render(
    <IssueBanner
      issue={{
        ...issue,
        body: thumbnailTomlString + issue.body,
      }}
    />,
  );

  expect(container).toMatchSnapshot();
});

it("썸네일 주소가 없을경우, 이슈 제목이 적힌 p 태그를 렌더링한다.", () => {
  render(<IssueBanner issue={issue} />);

  const p = screen.getByText(issue.title);

  expect(p.textContent).toBe(issue.title);
});
