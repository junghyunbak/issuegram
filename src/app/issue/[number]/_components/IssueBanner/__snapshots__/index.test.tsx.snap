// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`썸네일 주소가 있을경우, img 태그를 렌더링한다. 1`] = `
<div>
  <div
    class="flex h-full overflow-hidden max-md:aspect-square max-md:w-full w-[300px]"
  >
    <div
      class="relative flex size-full items-center justify-center"
    >
      <img
        class="absolute size-full object-cover"
        src="https://fake.url"
      />
      <div
        class="absolute size-full backdrop-blur"
      />
      <img
        class="absolute object-contain"
        src="https://fake.url"
      />
    </div>
  </div>
</div>
`;
